packages/ui/src/components/TaskCard/TaskCard.tsx

```diff
@@ -0,0 +1,138 @@
+import { cloneElement, ReactElement, ReactNode } from "react";
+
+import { IconType } from "../../types";
+import { LinkProps } from "../Link";
+import { HeadingLevel } from "../Heading";
+import { headingStyles } from "../Heading/styles";
+
+export const colorOptions = [
+  "primary",
+  "secondary",
+  "tertiary",
+  "quaternary",
+  "quinary",
+] as const;
+
+type CardColor = (typeof colorOptions)[number];
+
+const headingBarStyleMap: Record<CardColor, Record<string, string>> = {
+  primary: {
+    "data-h2-background-color": "base(primary.lightest)",
+    "data-h2-border-bottom": "base(primary.darkest)",
+  },
+  secondary: {
+    "data-h2-background-color": "base(secondary.lightest)",
+    "data-h2-border-bottom": "base(secondary.darkest)",
+  },
+  tertiary: {
+    "data-h2-background-color": "base(tertiary.lightest)",
+    "data-h2-border-bottom": "base(tertiary.darkest)",
+  },
+  quaternary: {
+    "data-h2-background-color": "base(quaternary.lightest)",
+    "data-h2-border-bottom": "base(quaternary.darkest)",
+  },
+  quinary: {
+    "data-h2-background-color": "base(quinary.lightest)",
+    "data-h2-border-bottom": "base(quinary.darkest)",
+  },
+};
+
+const wrapperStyleMap: Record<CardColor, Record<string, string>> = {
+  primary: {
+    "data-h2-color": "base(primary.darkest)",
+  },
+  secondary: {
+    "data-h2-color": "base(secondary.darkest)",
+  },
+  tertiary: {
+    "data-h2-color": "base(tertiary.darkest)",
+  },
+  quaternary: {
+    "data-h2-color": "base(quaternary.darkest)",
+  },
+  quinary: {
+    "data-h2-color": "base(quinary.darkest)",
+  },
+};
+
+export interface TaskCardProps {
+  icon?: IconType;
+  title: ReactNode;
+  headingColor?: CardColor;
+  link?: ReactElement<LinkProps>;
+  headingAs?: HeadingLevel;
+  children?: ReactNode;
+}
+
+const TaskCard = ({
+  icon,
+  title,
+  headingColor = "primary",
+  link: rawLink,
+  headingAs = "h3",
+  children,
+}: TaskCardProps) => {
+  // prepare link
+  const headingBarLink = rawLink
+    ? cloneElement(rawLink, {
+        color: headingColor,
+      })
+    : null;
+
+  // prepare icon element
+  const Icon = icon;
+
+  // prepare heading text element
+  const HeadingTextElement = headingAs;
+  const commonHeadingTextStyles = { ...headingStyles["h4"] };
```

Comment:
We should change the "h4" in `...headingStyles["h4"]` to the prop value `headingAs`.

Reply:
I'm not sure about this one.  I actually do want it to have all the styles of an h4.  
![image](https://github.com/user-attachments/assets/ca2d7a54-922e-4b05-afa6-0cd4290d4d6c)

Notice that semantic level is still headingAs.
![image](https://github.com/user-attachments/assets/69dcb6a9-f039-4cf7-99cc-9edfed3fbeb0)

Reply:
Oh actually you can use the <Heading /> component instead. That way you don't need to import the styles and delete the margin. It would look something like:

```tsx
<Heading
  level={headingAs}
  size="h4"
  data-h2-margin="base(0)"
  data-h2-text-align="base(center) p-tablet(left)"
>
  {title}
</Heading>
```

Reply:
Great idaea!
[use Heading component](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11161/commits/a1c5b895e83a0bd9c657f91e00eaecc256f0ce93)